src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +777,185 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * This overrides the add_data method from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
+			$args = $this->get_normalized_script_args( $handle, $value );
+			if ( $args['in_footer'] ) {
+				parent::add_data( $handle, 'group', 1 );
+			}
+			return parent::add_data( $handle, $key, $args );
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script() {
+		foreach ( $this->registered as $handle => $script ) {
+			// non standalone after scripts of async or defer are usually delayed.
+			if ( in_array( $this->get_intended_strategy( $handle ), array( 'defer', 'async' ), true ) &&
+				$this->has_non_standalone_inline_script( $handle, 'after' )
+			) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Normalize the data inside the $args parameter and support backward compatibility.
+	 *
+	 * @param string        $handle Name of the script.
+	 * @param array         $args     {
+	 *      Optional. Additional script arguments. Default empty array.
+	 *
+	 *      @type boolean   $in_footer    Optional. Default true.
```

Comment:
Looks like the default is actually `false`, per line 830. Does this doc also need to be updated for `wp_register_script` and `wp_enqueue_script`?
```suggestion
	 *      @type boolean   $in_footer    Optional. Default false.
```

Reply:
👍🏼 default should stay the same as current (`false`)

Reply:
Fixed at all the places.